twenty-first/src/shared_math/polynomial.rs

```diff
@@ -730,7 +470,11 @@ impl<PFElem: FiniteField> Polynomial<PFElem> {
     }
 
     /// The inverse of `fast_coset_evaluate`
-    pub fn fast_coset_interpolate(offset: &PFElem, generator: PFElem, values: &[PFElem]) -> Self {
+    pub fn fast_coset_interpolate(
+        offset: &BFieldElement,
+        generator: BFieldElement,
+        values: &[PFElem],
+    ) -> Self {
```

Comment:
I suppose this means that we don't generally `fast_coset_interpolate`.

Reply:
I think we only run it over `XFieldElement` codewords, but this code will also work for `BFieldElement` codewords. We always use an offset and a generator from the B field though. The faster NTT needs to know that the generator is a `BFieldElement` now, that change in NTT propagates up to many of the NTT-based polynomial functions.